crates/sequencer/src/commitment/strategy.rs

```diff
@@ -0,0 +1,177 @@
+    }
+}
+
+impl<Db: SequencerLedgerOps> CommitmentStrategy for MinSoftConfirmations<Db> {
+    fn should_commit(
+        &mut self,
+        l2_height: u64,
+        _l2_state_diff: StateDiff,
+    ) -> anyhow::Result<Option<CommitmentInfo>> {
+        // Get latest finalized and pending commitments and find the max height
+        let last_finalized_l2_height = self
+            .ledger_db
+            .get_last_commitment_l2_height()?
+            .unwrap_or(BatchNumber(0));
+        let last_pending_l2_height = self
+            .ledger_db
+            .get_pending_commitments_l2_range()?
+            .iter()
+            .map(|(_, end)| *end)
+            .max()
+            .unwrap_or(BatchNumber(0));
+        let last_committed_l2_height = cmp::max(last_finalized_l2_height, last_pending_l2_height);
+
+        // If the last commitment made is on par with the head
+        // soft confirmation, we have already committed the latest block.
+        if last_committed_l2_height.0 >= l2_height {
+            // Already committed.
+            return Ok(None);
+        }
+
+        let l2_start = last_committed_l2_height.0 + 1;
+        let l2_end = l2_height;
+
+        let l2_range_length = 1 + l2_end - l2_start;
+        if l2_range_length < self.number {
+            return Ok(None);
+        }
+
+        debug!("Enough soft confirmations to submit commitment");
+        Ok(Some(CommitmentInfo {
+            l2_height_range: BatchNumber(l2_start)..=BatchNumber(l2_end),
+        }))
+    }
+}
+
+pub(crate) struct StateDiffThreshold<Db>
+where
+    Db: SequencerLedgerOps,
+{
+    ledger_db: Arc<Db>,
+    last_state_diff: StateDiff,
+}
+
+impl<Db: SequencerLedgerOps> StateDiffThreshold<Db> {
+    pub fn new(ledger_db: Arc<Db>) -> Self {
+        let last_state_diff = ledger_db.get_state_diff().unwrap_or_default();
+        Self {
+            ledger_db,
+            last_state_diff,
+        }
+    }
+}
+
+impl<Db> CommitmentStrategy for StateDiffThreshold<Db>
+where
+    Db: SequencerLedgerOps,
+{
+    fn should_commit(
+        &mut self,
+        l2_height: u64,
+        l2_state_diff: StateDiff,
+    ) -> anyhow::Result<Option<CommitmentInfo>> {
+        // Get latest finalized and pending commitments and find the max height
+        let last_finalized_l2_height = self
+            .ledger_db
+            .get_last_commitment_l2_height()?
+            .unwrap_or(BatchNumber(0));
+        let last_pending_l2_height = self
+            .ledger_db
+            .get_pending_commitments_l2_range()?
+            .iter()
+            .map(|(_, end)| *end)
+            .max()
+            .unwrap_or(BatchNumber(0));
+        let last_committed_l2_height = cmp::max(last_finalized_l2_height, last_pending_l2_height);
+
+        let merged_state_diff =
+            merge_state_diffs(self.last_state_diff.clone(), l2_state_diff.clone());
+        let compressed_state_diff = compress_blob(&borsh::to_vec(&merged_state_diff)?);
```

Comment:
Can you bench how much it takes to compress? If thats a long cpu task, maybe we should spawn it in a blocking task?

Reply:
yes for big state diffs it might be slow

Reply:
It's done. We execute commitment controller strategies in the blocking thread.